tests/stats/test_main.py

```diff
@@ -0,0 +1,141 @@
+    charts_url = parse.urljoin(base_url, "/api/v1/chart")
+    datasets_url = parse.urljoin(base_url, "/api/v1/dataset")
+    database_conn_url = parse.urljoin(base_url, "/api/v1/database/test_connection")
+    db_type, db_user, db_password, db_host, db_port, db_name = (
+        getattr(args, "db.type"),
+        getattr(args, "db.user"),
+        getattr(args, "db.password"),
+        getattr(args, "db.host"),
+        getattr(args, "db.port"),
+        getattr(args, "db.name"),
+    )
+    sqla_url = f"{db_type}://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}"
+    database_data = {
+        "configuration_method": "sqlalchemy_form",
+        "database_name": "dff_database",
+        "driver": "string",
+        "engine": None,
+        "extra": "",
+        "impersonate_user": False,
+        "masked_encrypted_extra": "",
+        "parameters": {},
+        "server_cert": None,
+        "sqlalchemy_uri": sqla_url,
+        "ssh_tunnel": None,
+    }
+
+    session, headers = get_superset_session(args, base_url)
+    database_res = session.post(database_conn_url, json=database_data, headers=headers)
+    assert database_res.status_code == 200
+    dashboard_res = session.get(dashboard_url, headers=headers)
+    assert dashboard_res.status_code == 200
+    dashboard_json = dashboard_res.json()
+    assert dashboard_json["result"]["charts"] == [
+        "Flow visit ratio monitor",
+        "Node Visits",
+        "Node counts",
+        "Node visit ratio monitor",
+        "Node visits [cloud]",
+        "Node visits [ratio]",
+        "Node visits [sunburst]",
+        "Service load [max dialogue length]",
+        "Service load [users]",
+        "Table",
+        "Terminal labels",
+        "Transition counts",
+        "Transition layout",
+        "Transition ratio [chord]",
+    ]
+    assert dashboard_json["result"]["url"] == "/superset/dashboard/dff-stats/"
+    assert dashboard_json["result"]["dashboard_title"] == "DFF Stats"
+    datasets_result = session.get(datasets_url, headers=headers)
+    datasets_json = datasets_result.json()
+    assert datasets_json["count"] == 2
+    assert datasets_json["ids"] == [1, 2]
+    assert [item["id"] for item in datasets_json["result"]] == [1, 2]
+    assert sorted([item["table_name"] for item in datasets_json["result"]]) == [
+        "dff_final_nodes",
+        "dff_node_stats",
+    ]
+    charts_result = session.get(charts_url, headers=headers)
+    charts_json = charts_result.json()
+    assert charts_json["count"] == 14
+    assert sorted(charts_json["ids"]) == list(range(1, 15))
+    session.close()
+
+
+@pytest.mark.skipif(not SUPERSET_ACTIVE, reason="Superset server not active")
+@pytest.mark.parametrize(
+    ["args"],
+    [
+        (
+            Namespace(
+                **{
+                    "outfile": "1.zip",
+                    "db.type": "clickhousedb+connect",
+                    "db.user": os.getenv("CLICKHOUSE_USER"),
```

Comment:
Please, consider (in all tests and tutorials) using `os.environ[...]` if returning `None` is not expected.

Reply:
Done